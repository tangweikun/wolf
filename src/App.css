.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40px;
}

.App-header {
  position: fixed;
  left: 200px;
  width: 100%;
  background-color: #222;
  height: 40px;
  padding: 15px;
  color: white;
  z-index: 9999;
}

.App-body {
  position: relative;
  left: 200px;
  top: 70px;
  width: calc(100% - 200px);
}

.App-sidebar {
  width: 200px;
  position: fixed;
  height: 100%;
  background-color: rgb(48, 48, 48);
  left: 0;
  top: 0;
}

.App-sidebar-title {
  background-color: rgba(0, 0, 0, .87);
  color: #fff;
  height: 70px;
  line-height: 70px;
  font-size: 24px;
  padding-left: 20px;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
