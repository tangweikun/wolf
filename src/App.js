import React, { Component } from 'react';
import MenuItem from 'material-ui/MenuItem';
import Menu from 'material-ui/Menu';
import { browserHistory } from 'react-router';
import logo from './logo.svg';
import './App.css';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = { title: 'Keep Learning' };
  }
  render() {
    const sidebarList = [
      { name: '读书记录', path: '/about' },
      { name: '体重监测', path: '/demo' },
      { name: 'CSS漫展', path: '/comicon' },
      { name: '小黄人', path: '/minion' },
      { name: 'Lodash', path: '/' },
      { name: '资产', path: '/property' },
    ];

    return (
      <div className="App">
        <div className="App-sidebar">
          <div className="App-sidebar-title">KeepLearning</div>
          <Menu onItemTouchTap={() => { console.log(454); }}>
            {
                sidebarList.map(({ name, path }) =>
                  (<MenuItem
                    key={name}
                    onTouchTap={() => {
                      browserHistory.push(path);
                      this.setState({ title: name });
                    }}
                  >
                    {name}
                  </MenuItem>),
                )
              }
          </Menu>
        </div>
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <label>{this.state.title}</label>
        </div>
        <div className="App-body">
          {this.props.children}
        </div>
      </div>
    );
  }
}
