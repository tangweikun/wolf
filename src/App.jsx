import React, { Component } from 'react'
import MenuItem from 'material-ui/MenuItem'
import Menu from 'material-ui/Menu'
import { browserHistory } from 'react-router'
import PropTypes from 'prop-types'

import logo from './logo.svg'
import './App.css'

export default class App extends Component {
  constructor(props) {
    super(props)
    this.state = { title: 'TodoList' }
  }

  render() {
    const sidebarList = [
      { name: 'TodoList', path: '/todo' },
      { name: 'TaskList', path: '/taskList' },
    ]

    return (
      <div className="App">
        <div className="App-sidebar">
          <div className="App-sidebar-title">KeepLearning</div>
          <Menu>
            {sidebarList.map(({ name, path }) =>
              <MenuItem
                key={name}
                style={{ color: '#fff', width: '200px' }}
                onTouchTap={() => {
                  browserHistory.push(path)
                  this.setState({ title: name })
                }}
              >
                {name}
              </MenuItem>,
            )}
          </Menu>
        </div>
        <div>
          <div className="App-header">
            <img src={logo} className="App-logo" alt="logo" />
            <span>
              {this.state.title}
            </span>
          </div>
          <div className="App-body">
            {this.props.children}
          </div>
        </div>
      </div>
    )
  }
}

App.propTypes = {
  children: PropTypes.element.isRequired,
}
