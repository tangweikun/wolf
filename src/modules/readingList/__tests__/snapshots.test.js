import React from 'react'
import renderer from 'react-test-renderer'

import App from '../components/index.jsx'

describe('++ Snapshot for Test', () => {
  it('should render TableRow', () => {
    const tree = renderer.create(<App />).toJSON()
    // const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
