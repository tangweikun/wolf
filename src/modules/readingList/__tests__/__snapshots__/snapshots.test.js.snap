// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`++ Snapshot for Test should render TableRow 1`] = `
<div>
  <div>
    <div>
      <span>
        React引领未来的用户界面开发框架
      </span>
      <span>
        twk
      </span>
      <span>
        1467504000000
      </span>
    </div>
    <div>
      <span>
        JavaScript面向对象编程指南
      </span>
      <span>
        twk2
      </span>
      <span>
        1470182400000
      </span>
    </div>
    <div>
      <span>
        ES6标准入门
      </span>
      <span>
        ruanyifeng
      </span>
      <span>
        1470182400000
      </span>
    </div>
  </div>
  <input
    onChange={[Function]}
    value={undefined}
  />
  <p />
</div>
`;
